### DISCUSSION TITLE
Volt Lifecycle Hooks

### DISCUSSION THREAD

**Discussion by citricguy** [livewire/livewire#7092] [1. Help]
How can I convert the following class based lifecycle hook to use Volt's functional lifecycle hooks?

```
public function updated($property, $value)
    {
        $this->validateOnly($property);
    }
```

I can get things to work for a specific field but cannot seem to get it to work with a variable like above.
`updated(['name' => fn () => $this->validateOnly('name')]);`

Here is more or less what I would like to work, but it fails due to $fieldName being undefined:
`updated([$fieldName => fn () => $this->validateOnly($fieldName)]);`

Do I need to switch to class based over functional, or is there something I'm missing?

**Comment by Sindoweb** [comment]
This is more or less with what I'm struggling with.

I have multiple selects with a dynamic name:

`<select wire:model.live="changeColumns.{{ $content->id }}" >
                                @for($i = 2; $i <= 12; $i++)
                                    <option value="{{ $i }}">{{ $i }}</option>
                                @endfor
                            </select>`

The values are being filled in the mount hook - no issue there.
But how do I go about the updated hook when a dropdown changes?

This works, but it's obviously not dynamic.

`updated([
    'changeColumns.1' => function ($value) {
       dd('test');
    }
]);`
